creator/data_reviews/mutations.py

```diff
@@ -0,0 +1,323 @@
+from django_fsm import TransitionNotAllowed
+import graphene
+from graphql import GraphQLError
+from graphql_relay import from_global_id
+from django.db import transaction
+
+from creator.data_reviews.nodes import DataReviewNode
+from creator.studies.models import Study
+from creator.data_reviews.models import DataReview, State
+
+
+def check_review_files(input, study):
+    """
+    Check if files exist and they are from same study
+    """
+    review_file_ids = []
+    if input.get("files"):
+        study_file_ids = {f.pk for f in study.files.all()}
+        for file_ in input["files"]:
+            _, file_id = from_global_id(file_)
+            if file_id not in study_file_ids:
+                raise GraphQLError(
+                    f"Error in modifying data_review. All files in data "
+                    "review must come from the same study. File "
+                    f"{file_id} does not exist in study {study.pk}"
+                )
+            review_file_ids.append(file_id)
+    return review_file_ids
+
+
+class CreateDataReviewInput(graphene.InputObjectType):
+    """ Parameters used when creating a new data_review """
+
+    name = graphene.String(
+        required=True, description="The name of the data_review"
+    )
+    description = graphene.String(
+        required=True, description="The description of the data_review"
+    )
+    study = graphene.ID(
+        required=True,
+        description="The ID of the study this data_review is for",
+    )
+    files = graphene.List(
+        graphene.ID,
+        description="Files in this data_review",
+    )
+
+
+class UpdateDataReviewInput(graphene.InputObjectType):
+    """ Parameters used when updating an existing data_review """
+
+    name = graphene.String(description="The name of the data_review")
+    description = graphene.String(
+        description="The description of the data_review"
+    )
+    files = graphene.List(
+        graphene.ID,
+        description="Files in this data_review",
+    )
+
+
+class CreateDataReviewMutation(graphene.Mutation):
+    """ Creates a new data_review """
+
+    class Arguments:
+        input = CreateDataReviewInput(
+            required=True, description="Attributes for the new data_review"
+        )
+
+    data_review = graphene.Field(DataReviewNode)
+
+    def mutate(self, info, input):
+        """
+        Creates a new data_review.
+        """
+        user = info.context.user
+
+        # Check if study exists
+        _, study_id = from_global_id(input["study"])
+        try:
+            study = Study.objects.get(pk=study_id)
+        except Study.DoesNotExist:
+            raise GraphQLError(
+                f"Error in creating data_review. Study {study_id} not found."
```

Comment:
I think the 'Error in creating data_review.' is omitted in all our other not found responses.